Services/service_response.c

```diff
@@ -32,59 +32,67 @@ extern Service_Queues_t service_queues;  // Implemented by the host platform
 /**
  * @brief
  * 		Wait on a queue of responses to be sent to other CSP nodes
- * (usually the ground)
+ *              (usually the ground)
  * @details
  * 		CSP client server will wake when data is in the queue for
- * downlink (telemetery)
+ *              downlink (telemetery)
  * @param void * param
  * 		Not used
  * @return
- * 		should not return
+ * 		A SAT_returnState depending on whether we were successful
+ *              in connecting. Not typically used for anything.
  */
-void service_response_task(void *param) {
+SAT_returnState service_response_task(void *param) {
   TC_TM_app_id my_address = SYSTEM_APP_ID;
   csp_packet_t *packet;
-  uint32_t in;
+  uint32_t data;
   for (;;) {
-    /* To get conn from the response queue */
+
     if (xQueueReceive(service_queues.response_queue, &packet,
                       NORMAL_TICKS_TO_WAIT) == pdPASS) {
-      /* For some reason, if we directly print packet->data[DATA_BYTE],
-         it will be set to arg we sent.
-      */
-//      printf("Set to %u\n",packet->data[DATA_BYTE]);
-      //cnv8_32(&packet->data[DATA_BYTE], &in);
-      //printf("Set to %u\n", (uint32_t)in);
+      cnv8_32(&packet->data[DATA_BYTE], &data);
+      printf("Packet data before sending: %u\n", (uint32_t) data);
+      
+      uint8_t dest_addr = GND_APP_ID; // Change this as needed!
+      uint8_t dest_port = (uint8_t) packet->data[SUBSERVICE_BYTE]; // The desination subservice
+      csp_log_info("Sending to service %u and subservice %u", (uint32_t) dest_addr, (uint32_t) dest_port);
+
+      // Connect with a connectionfull method
+      // csp_connect(priority, desination address, dest. port, timeout (ms), options);
+      csp_conn_t *conn = csp_connect(CSP_PRIO_NORM, dest_addr, dest_port, 1000, CSP_O_NONE);
       
+      if (conn == NULL) {
+	csp_log_error("Failed to get CSP CONNECTION");
```

Comment:
Indentation

Reply:
Thank you; the emacs config was indenting by tabs there.